### PR TITLE
[Notebooks] Add Hunter as a Regression Indicator

### PR DESCRIPTION
**Pull request by drewrip**

### CODE REVIEW

**Comment by drewrip** [comment]
/test rhoai-plot notebooks https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/195/pull-ci-openshift-psap-topsail-main-rhoai-light/1753099883524395008/artifacts/light/test/artifacts/002__sutest_notebooks__benchmark_performance/
/var matbench.workload=rhods-notebooks-performance
/var matbench.config_file=single_notebook_perf.yaml
/var matbench.lts.opensearch.index=regression-test.rhoai-notebooks-performance
/var matbench.lts.opensearch.upload=false

**Comment by drewrip** [comment]
/test rhoai-plot notebooks https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/195/pull-ci-openshift-psap-topsail-main-rhoai-light/1753099883524395008/artifacts/light/test/artifacts/002__sutest_notebooks__benchmark_performance/
/var matbench.workload=rhods-notebooks-performance
/var matbench.config_file=single_notebook_perf.yaml
/var matbench.lts.opensearch.index=regression-test.rhoai-notebooks-performance
/var matbench.lts.opensearch.upload=false

**Comment by drewrip** [comment]
/test rhoai-plot notebooks https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/195/pull-ci-openshift-psap-topsail-main-rhoai-light/1753099883524395008/artifacts/light/test/artifacts/002__sutest_notebooks__benchmark_performance/
/var matbench.workload=rhods-notebooks-performance
/var matbench.config_file=single_notebook_perf.yaml
/var matbench.lts.opensearch.index=regression-test.rhoai-notebooks-performance
/var matbench.lts.opensearch.upload=false

**Comment by drewrip** [comment]
[regression analysis results](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/200/pull-ci-openshift-psap-topsail-main-rhoai-plot/1757506068025446400/artifacts/plot/test/artifacts/report_02_report:%20LTS%20Documentation.html) successfully generated using hunter

File: projects/notebooks/testing/generate_matrix-benchmarking.sh [hunk 1]
```diff
@@ -21,309 +21,310 @@ export MATBENCH_WORKLOAD=projects.notebooks.visualizations.$(get_config matbench
...
+			fi
+		fi
+
+		rm -f .env.yaml
+		step_idx=$((step_idx + 1))
+		retcode=0
+		if ! matbench parse --output_lts $ARTIFACT_DIR/lts_payload.json |& tee >"$ARTIFACT_DIR/${step_idx}_matbench_generate_lts.log"; then
+			_warning "An error happened while generating the LTS payload from $MATBENCH_RESULTS_DIRNAME :/."
+			retcode=1
+		fi
```
Comment (kpouget) [inline projects/notebooks/testing/generate_matrix-benchmarking.sh:230]: I'm not sure why you duplicated this block? 🤔 
I think the LTS payload shouldn't change between the first call to `parse --output-lts` and now 🤔

File: projects/notebooks/testing/generate_matrix-benchmarking.sh [hunk 2]
```diff
@@ -21,309 +21,310 @@ export MATBENCH_WORKLOAD=projects.notebooks.visualizations.$(get_config matbench
...
+	set_config matbench.config_file "${matbench_preset}.yaml"
+	set_config matbench.download.url_file "${WORKLOAD_STORAGE_DIR}/data/${matbench_preset}.yaml"
 fi
 
 get_matbench_config() {
-    CI_ARTIFACTS_FROM_CONFIG_FILE=$WORKLOAD_STORAGE_DIR/data/$(get_config matbench.config_file) \
-        get_config "$@"
+	CI_ARTIFACTS_FROM_CONFIG_FILE=$WORKLOAD_STORAGE_DIR/data/$(get_config matbench.config_file) \
+		get_config "$@"
 }
```
Comment (kpouget) [inline projects/notebooks/testing/generate_matrix-benchmarking.sh:44]: in this file, could you revert the indentation to 4 spaces, and add the few empty lines back? (or are these empty lines inconsistent with the rest of the file? 🤔 )

File: projects/notebooks/testing/generate_matrix-benchmarking.sh [hunk 3]
```diff
@@ -21,309 +21,310 @@ export MATBENCH_WORKLOAD=projects.notebooks.visualizations.$(get_config matbench
...
+	# Analyze the current results against LTS results
+	#
+
+	if test_config matbench.lts.regression_analyses.enabled; then
+		generate_opensearch_config
+
+		step_idx=$((step_idx + 1))
+
+		if ! matbench download_lts --lts_results_dirname "$MATBENCH_RESULTS_DIRNAME/lts" |&
+			tee >"$ARTIFACT_DIR/${step_idx}_matbench_download_lts.log"; then
```
Comment (kpouget) [inline projects/notebooks/testing/generate_matrix-benchmarking.sh:197]: not sure why you changed this,
I like *not* to have the `|`, `|&` or `||` at the end of the line, the idea is to keep important information/operators at the beginning of the line, as it's critical to understand the relationship between the different lines

same for the `then`, if the if condition has multiple lines, I move the `then` to a dedicated line, to clearly distinguish the condition from the `if` statement


```
if very long condition; then
    first statement
```

```
if long condition number 1 &&
    long condition number 2; then
    first statement
```
in these ^^^, if you hide the last ~10 chars, you can't distinguish the conditions from the statements
```
if long condition 1 \
   && long condition 2;
then
    first statement
```
with this, you have 2 ways to distinguish them

File: projects/notebooks/testing/generate_matrix-benchmarking.sh [hunk 4]
```diff
@@ -21,309 +21,310 @@ export MATBENCH_WORKLOAD=projects.notebooks.visualizations.$(get_config matbench
...
+			_warning "An error happened while downloading the LTS results :/."
+			retcode=1
+		fi
+
+		step_idx=$((step_idx + 1))
+		regression_analyses_retcode=0
+		matbench analyze-lts --lts_results_dirname "$MATBENCH_RESULTS_DIRNAME/lts" |&
+			tee >"$ARTIFACT_DIR/${step_idx}_matbench_analyze_lts.log" ||
```
Comment (kpouget) [inline projects/notebooks/testing/generate_matrix-benchmarking.sh:205]:
```suggestion
			tee > "$ARTIFACT_DIR/${step_idx}_matbench_analyze_lts.log" ||
```
I try to always keep a space before/after the `>` (except if redirecting into a file description, like `>&2`), let's avoid changing it here

File: projects/notebooks/visualizations/rhods-notebooks-performance/store/parsers.py [hunk 5]
```diff
@@ -258,7 +259,7 @@ def _parse_start_end_time(dirname):
 def _parse_regression_results(dirname):
     regression_results_file = dirname / "regression.json"
     if not regression_results_file.exists():
-        logging.info(f"{regression_results_file.name} does not exist, ignoring the parsing of the regression analyses results.")
+        logging.info(f"{regression_results_file} does not exist, no new regression results found.")
```
Comment (kpouget) [inline projects/notebooks/visualizations/rhods-notebooks-performance/store/parsers.py:262]: can you remove this line (or turn it into a `debug` call)?
or perhaps move it after the `return None` and turn it positive

as it is (from my code), it's too chatty and does not help much

File: projects/notebooks/visualizations/rhods-notebooks-performance/store/parsers.py [hunk 6]
```diff
@@ -298,6 +299,7 @@ def _parse_env(dirname, test_config):
     else:
         from_env.test.test_path = str((current_artifact_dir / dirname).relative_to(base_artifact_dir))
 
+    outside_test_env = os.getenv("OUTSIDE_TEST_ENV")
```
Comment (kpouget) [inline projects/notebooks/visualizations/rhods-notebooks-performance/store/parsers.py:302]: hum, I see what happened. Can you remove this line?

File: projects/notebooks/visualizations/rhods-notebooks-performance/store/parsers.py [hunk 7]
```diff
@@ -343,7 +345,7 @@ def _parse_env(dirname, test_config):
             JENKINS_ARTIFACTS=f"https://{jenkins_instance}/{jenkins_job}/{build_number}/artifact/run/{jumphost}/{base_path}/{from_env.test.test_path}"
         )
 
-    if test_config.get("export_artifacts.enabled"):
+    if test_config.get("export_artifacts.enabled") and not outside_test_env:
```
Comment (kpouget) [inline projects/notebooks/visualizations/rhods-notebooks-performance/store/parsers.py:348]: can you turn this into:
```suggestion
    if test_config.get("export_artifacts.enabled") and from_env.test.ci_engine is not None:
```

and maybe above, replace:
```
if ansible_env.get("PERFLAB_CI") == "true":
by
if from_env.test.ci_engine == "PERFLAB_CI"
and same above:
if from_env.test.ci_engine == "OPENSHIFT_CI"
```

File: projects/notebooks/testing/generate_matrix-benchmarking.sh (hunk 1 above)
Comment (kpouget) [reply projects/notebooks/testing/generate_matrix-benchmarking.sh:230]: hum, got it. Can you add a comment:
```
# regenerate the LTS payload to capture the regression analyses results
```
and we'll have to update [the Python file](https://github.com/openshift-psap/topsail/blob/main/topsail/testing/visualize.py#L373) as well, eventually

**Review by kpouget — CHANGES_REQUESTED** [review]
thanks @drewrip , looks quite good, just a few nits to change

**Comment by kpouget** [comment]
closing as outdated

**Pull request closed without merging**